compiler/src/dotty/tools/dotc/transform/MoveStatics.scala

```diff
@@ -49,8 +59,24 @@ class MoveStatics extends MiniPhase with SymTransformer {
             staticCostructor.addAnnotation(Annotation(defn.ScalaStaticAnnot, staticCostructor.span))
             staticCostructor.entered
 
-            val staticAssigns = staticFields.map(x => Assign(ref(x.symbol), x.rhs.changeOwner(x.symbol, staticCostructor)))
-            tpd.DefDef(staticCostructor, Block(staticAssigns, tpd.unitLiteral)) :: newBody
+            val symbolRemap = localStaticDefs.map { x =>
```

Comment:
IMO this is too complex to introduce without a very good reason.

I suggest we stick to the simpler alternative where we repeat the calls to `lookup()`.